webapp/channels/src/components/widgets/modals/components/react_select_item.tsx

```diff
@@ -30,13 +31,33 @@ type Props = BaseSettingItemProps & {
     handleChange: (selected: OnChangeValue<Option, boolean>) => void;
 }
 
+// Function to extract text from FormattedMessage components
+export const getOptionLabel = (option: Option, intl: ReturnType<typeof useIntl>) => {
+    if (typeof option.label === 'string') {
+        return option.label;
+    }
+
+    // For FormattedMessage components, extract the defaultMessage and id
+    if (React.isValidElement(option.label)) {
```

Comment:
Extracting the message from a `FormattedMessage` feels not great since that seems like it's messing with React internals and possibly something that'll break down the line if someone passes a different element as one of these options.

How much work would it be to change it the type of `option.label` to be `string | MessageDescriptor`? `MessageDescriptor`s are one of those objects with an `id` and a `defaultMessage`, and we have pass those around most places where we need translated text in string form. If you did that, you could also use `formatAsString` from `utils/i18n` here. You'd need to use `defineMessage` where those options are defined though for them to be picked up by the i18n tooling

Reply:
@hmhealey  The usage of the ReactSelect is most common with the `FormattedMessage` but agreed it currently doesn't guard this properly here. Maybe adding an additional check would alleviate that concern without having to change all the text components options passed to the comboboxes. ?
```suggestion
    if (React.isValidElement(option.label) && 
        option.label.type === FormattedMessage) {
```

Reply:
I don't think this type is actually used everywhere that React Select is used. If I'm searching through the code right, this is only used in the `ChannelNotificationsModal`, so it's only that one place that would need to be updated

Reply:
I can change it but again if others use it with passing <FormatMessage> then it won't work either. We can support both as well - again, the lack of consistency of usage is not ideal.

Reply:
@hmhealey finally got back to this - it's now updated with this feedback. Much cleaner now

Reply:
Thanks for changing it! It does look super clean now